web-console/src/views/datasource-view/datasource-view.tsx

```diff
@@ -144,42 +144,41 @@ function progress(done: number, awaiting: number): number {
 
 const PERCENT_BRACES = [formatPercent(1)];
 
-interface Datasource {
-  datasource: string;
-  rules: Rule[];
-  compactionConfig?: CompactionConfig;
-  compactionStatus?: CompactionStatus;
-  [key: string]: any;
+interface DatasourceQueryResultRow {
+  readonly datasource: string;
+  readonly num_segments: number;
+  readonly num_available_segments: number;
+  readonly num_segments_to_load: number;
+  readonly num_segments_to_drop: number;
+  readonly total_data_size: number;
+  readonly replicated_size: number;
+  readonly min_segment_size: number;
+  readonly avg_segment_size: number;
+  readonly max_segment_size: number;
```

Comment:
nit: should this be `min_segment_rows` ... instead of size?

Reply:
fixed